### DISCUSSION TITLE
Raw sql queries

### DISCUSSION THREAD

**Discussion by PhilipMur** [oqtane/oqtane.framework#1122] [Q&A] [answered]
This might be a question already answered but I could not find it in the discussion section.

Can I pass a raw sql query from an external module using an sql client etc. 

I don't use Entity framework or Daper etc because most of the time  the databases I work with could change from time to time and I find it easy to get the query right myself rather than relying on external tools. 

I'm pretty new to Blazor and Oqtane so I'm not familiar yet with how it all flows together.

**Comment by hishamco** [comment]
EF allows you to execute raw SQL, but in case of Oqtane, all the services and manages provide a rich APIs to deal with, i think `v2.1.0` as @sbwalker refer to will support mutiple databases, which is useful in your case

**Comment by sbwalker** [comment, marked as answer]
@PhilipMur if you are build custom modules you are free to use whatever data access approach you prefer. That being said, if you use EF Core it will simplify your efforts significantly because it provides the infrastructure to support multi-tenancy, and there are plenty of functional examples to help guide you. If you truly want to write your own SQL queries I would suggest you still use the DbContextBase approach and then utilize raw sql in your methods as described here: https://www.learnentityframeworkcore.com/raw-sql.

**Comment by PhilipMur** [comment]
Thanks for the help.
